index.bs

```diff
@@ -2210,22 +2211,84 @@ partial interface MLGraphBuilder {
   MLOperand gemm(MLOperand a, MLOperand b, optional MLGemmOptions options = {});
 };
 </script>
-<div algorithm=gemm>
+
+{{MLGemmOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLGemmOptions>
+    : <dfn>c</dfn>
+    ::
+        An {{MLOperand}}. Specifies the third input tensor. It is either a scalar, or of the shape that is unidirectionally broadcastable to the shape [M, N] according to [[!numpy-broadcasting-rule]]. When it is not specified, the computation is done as if *c* is a scalar `0.0`.
+
+    : <dfn>alpha</dfn>
+    ::
+        A {{float}} scalar multiplier for the first input.
+
+    : <dfn>beta</dfn>
+    ::
+        A {{float}} scalar multiplier for the third input {{MLGemmOptions/c}}.
+
+    : <dfn>aTranspose</dfn>
+    ::
+        A {{boolean}} indicating if the first input should be transposed prior to calculating the output.
+
+    : <dfn>bTranspose</dfn>
+    ::
+        A {{boolean}} indicating if the second input should be transposed prior to calculating the output.
+</dl>
+
+<div>
     **Arguments:**
         - *a*: an {{MLOperand}}. The first input 2-D tensor with shape [M, K] if *aTranspose* is false, or [K, M] if *aTranspose* is true.
         - *b*: an {{MLOperand}}. The second input 2-D tensor with shape [K, N] if *bTranspose* is false, or [N, K] if *bTranspose* is true.
         - *options*: an optional {{MLGemmOptions}}. The optional parameters of the operation.
-            - *c*: an {{MLOperand}}. The third input tensor. It is either a scalar, or of the shape that is unidirectionally broadcastable to the shape [M, N] according to [[!numpy-broadcasting-rule]]. When it is not specified, the computation is done as if *c* is a scalar 0.0.
-            - *alpha*: a {{float}} scalar multiplier for the first input, default to 1.0.
-            - *beta*: a {{float}} scalar multiplier for the third input, default to 1.0.
-            - *aTranspose*: a {{boolean}} indicating if the first input should be transposed prior to calculating the output, default to false.
-            - *bTranspose*: a {{boolean}} indicating if the second input should be transposed prior to calculating the output, default to false.
 
     **Returns:** an {{MLOperand}}. The output 2-D tensor of shape [M, N] that contains the calculated product of all the inputs.
+</div>
 
-    <div class="note">
+<details open>
+  <summary>
+    The {{MLGraphBuilder/gemm(a, b, options)}} steps are:
+  </summary>
+  <div algorithm=gemm class=algorithm-steps>
+    1. If |a| or |b| is not an instance of {{MLOperand}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
```

Comment:
Throw a DataError if the data type of |a| and |b| is not the same.

Reply:
Throughout the spec we have been using TypeError for this. It means should we change that to DataError? I wonder should we make an issue from this and discuss more widely, or should I just change to DataError in all such cases?
@huningxin WDYT?

Reply:
Drive by comment: the first step here (_If a or b is not an instance of MLOperand, then throw a "TypeError"..._)  is unnecessary, and I see this repeated throughout the newly introduced algorithms for methods ("operations" in Web IDL). The Web IDL bindings already take care validating input types and throwing TypeError.

Reply:
Further, even in cases where an explicit error must be thrown, the intrinsic type TypeError should be thrown, not a DOMException of type "TypeError" (which is not present in https://webidl.spec.whatwg.org/#idl-DOMException-error-names)

Reply:
Thanks Joshua, I will update according to this throughout the spec.